automate_apex_legends_shader_merge.sh

```diff
@@ -0,0 +1,128 @@
+
+# Print help
+
+if [[ "$1" == "-h" || $# -le 2 ]]; then
+    echo "Usage: ";
+    echo "      $0 STEAM_APEX_GAME_DIR LOCAL_REPO_DIR [MERGE_TOOL_DIR]";
+    echo;
+    echo "Best way to find STEAM_APEX_GAME_DIR is to open steam, go to library, right click on Apex Legends and select Manage->Browse local files. Your file browser should open up and you can copy the path to the Apex Legends game directory and supply it to this script."
+    echo;
+    echo "$manualSteps";
+    exit 0;
+fi
+
+gamefilepth=$1;
+clonedrepo=$2;
+mergetoolpth=$3;
+downloadaddr="https://lutris.nyc3.cdn.digitaloceanspaces.com/games/overwatch/merge-tool.tar.xz";
+downloaddir="$HOME";
+downloadfilename="merge-tool.tar.xz";
+mergeTooldir="$downloaddir/merge-tool";
+mergeToolLog="$HOME/mergeToolLog.log";
+
+if [[ ! -d $gamefilepth ]];
+then
+    echo "Game file path does not exist. Exit now!";
+    exit 1;
+fi
+
+if [[ ! -d $clonedrepo ]];
+then
+    echo "Cloned repo path does not exist. Exit now!";
+    exit 1;
+fi
+
+
+# Check if path to merge tool directory was provided on CLI.
+# If not, then ask if it should be downloaded. If download is denied, then exit.
+
+if [[ ! -d $mergetoolpth ]]; then
+    echo "Merge tool path does not exist or was not provided.";
+    read -p "Download merge tool now? [yes/no]: " isDownload;
+    if [[ $isDownload == "yes"  ]]; then
+        echo "Downloading now!";
+        retCode=$(curl -sSl -w "%{http_code}" -o "$downloaddir/$downloadfilename" $downloadaddr);
+        if [[  "$retCode" =~ ^2 ]]; then
+            echo "Download successful!";
+            if [[ ! -d $mergeTooldir ]]; then
+                mkdir $mergeTooldir;
+            fi
+            tar -xf "$downloaddir/$downloadfilename" -C $mergeTooldir;
+        elif [[ "$retCode" = 404 ]]; then
+            echo "Download failed!";
+            echo "Server returned 404. Exit now!";
+            exit 1;
+        else
+            echo "ERROR: server returned HTTP code $retCode";
+            exit 1;
+        fi
+    else
+        echo "Provide merge tool path. Exit now!"
+        exit 1;
+    fi
+fi
+
+shaderfilename="r5apex.dxvk-cache";
+searchstring="Steam/steamapps";
+relshaderpth="/shadercache/1172470/DXVK_state_cache";
+steamappspth="$(echo $gamefilepth | grep -o -E '.*(Steam\/steamapps)')";
+
+if [[ -z "$steamappspth" ]];    # Parse steam cache file directory
+then
+   echo "Could not find correct shader path. Exit now!";
+   exit 1;
+fi
+shaderpth="$steamappspth$relshaderpth";
+if [[ -d $shaderpth ]];
+then
+    echo "Found shader path:";
+    echo $shaderpth;
+fi
+
+if [[ -e "$shaderpth/$shaderfilename" ]]; then
+    cp -i "$shaderpth/$shaderfilename" "$clonedrepo/local-$shaderfilename";
+else
+    echo "Shader file does not exist. Exit now!";
+    exit 1;
+fi
+
+# TODO: Check if git and git lfs is installed
+touch "$mergeToolLog";
+cd "$clonedrepo";
+git status > "$mergeToolLog";
```

Comment:
To answer your question, it does not provide any useful information in the way it is currently used.
I used it as a quick and dirty sanity check. To see if using git like this works. Later I thought I could maybe use it to Log stuff that goes wrong (like not finding certain paths or missing access rights) so that the user could take a look at the log and maybe figure out what went wrong himself. Or use git's output to parse useful information. But I then decided, that before putting any more effort into "optimizing" a working script I would rather push and PR it to get some input from other people.

Also appending instead of overwriting would have been better, true. But it would have required me to add some kind of timestamp incase of multiple script executions, so that I could discern the different executions but at that time I was only interested on this single part of the script. Again quick and dirty....  I think I was also a bit tired and again tried to overcomplicate unneccessary things 😄

Reply:
Ah, okay. Thank you for that, thought I might be missing something.